lib/trample/search.rb

```diff
@@ -166,6 +166,23 @@ def records!
       records
     end
 
+    #This implementation is not using scroll and scan of elastic
+    #https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
+    #In future versions we hope to replace this with elastic in-built implementation
+    #
+    def find_in_batches(batch_size = 10_000)
+      page_number = 1
+
+      loop do
+        paginate(size: batch_size, number: page_number)
+        query!
+        yield results
+        offset = metadata.pagination.current_page * metadata.pagination.per_page
```

Comment:
Where is this variable actually used?

Now that I see an `offset` variable, this is a good option for `find_in_batches` to accept. A good example is bulk processing that fails mid-way through that you want to resume. But we can tackle this in a future PR sometime.